client/lxd_server.go

```diff
@@ -186,3 +187,50 @@ func (r *ProtocolLXD) GetMetrics() (string, error) {
 
 	return string(content), nil
 }
+
+// GetMetadataConfiguration returns metadata configuration for a server.
+func (r *ProtocolLXD) GetMetadataConfiguration() (*map[string]any, error) {
+	// Check that the server supports it.
+	err := r.CheckExtension("metadata_configuration")
+	if err != nil {
+		return nil, err
+	}
+
+	// Prepare the request.
+	requestURL, err := r.setQueryAttributes(fmt.Sprintf("%s/1.0/metadata/configuration", r.httpBaseURL.String()))
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("GET", requestURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Send the request.
+	resp, err := r.DoHTTP(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() { _ = resp.Body.Close() }()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Bad HTTP status: %d", resp.StatusCode)
+	}
+
+	// Get the content.
+	content, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal content.
+	config := make(map[string]any)
+	err = json.Unmarshal(content, &config)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I think you can use `r.queryStruct` here:
```go
config := make(map[string]any)
_, err = r.queryStruct(http.MethodGet, api.NewURL("metadata", "configuration").String(), nil, "", &config)
if err != nil {
    return err
}

return config, nil
```

Also note that this function will extract only the metadata from the request body. Your method here is returning the full API response as a `*map[string]any` which we don't need. The `queryStruct` function will return an error if the server returns an error.

Lastly, there's no need to return a `*map[string]any` as a map is already a pointer, can you please just return `map[string]any`, thanks.
```